### PR TITLE
Practice/java log4j configured

### PR DESCRIPTION
**Pull request by ryzzaki**
## Description
This PR aims to add a new practice that checks for the existence of configuration files for Log4J & it reads the files to check whether they are being configured using certain keys when they match.

**Note:** The reason why **__MOCKS__** aren't created in their own dedicated `.ts` files in the tests is because the Mocks themselves are very long in terms of lines of code, and it significantly reduces the Code Coverage %. Therefore, I decided to code it and place them directly in tests as this is something that we will _predictably_ use only once in this particular test (I included only one version of the mock in an `xml` format - should it ever be needed for a different purpose).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly.
- [x] I have added new practice to practice list in README.md.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/practices/Java/JavaLog4JConfiguredPractice.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          let parsedContents = {};
+          try {
+            switch (file.extension) {
+              case '.xml':
+                parsedContents = await xml2js.parseStringPromise(fileContents);
+                break;
+              case '.json':
+                parsedContents = await JSON.parse(fileContents);
+                break;
+              case '.yaml':
```
Comment (adelkahomolova) [inline src/practices/Java/JavaLog4JConfiguredPractice.ts]: What about case `.yml` ?

Comment (prokopsimek) [reply src/practices/Java/JavaLog4JConfiguredPractice.ts]: 👍 @ryzzaki 🙏

Comment (ryzzaki) [reply src/practices/Java/JavaLog4JConfiguredPractice.ts]: @prokopsimek @adelkahomolova sorry about that, it's fixed using a case fall-through and I added one more test to cover it :) I also reformated the internal error message.

**Pull request merged**